cognite/powerops/resync/v2/utils.py

```diff
@@ -0,0 +1,318 @@
+# Mypy does not understand the pydantic classes that allows both alias and name to be used in population
+# https://github.com/pydantic/pydantic/issues/3923
+# mypy: disable-error-code="call-arg"
+from __future__ import annotations
+
+import logging
+import random
+import re
+from pathlib import Path
+from typing import Optional, Union
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes import FileMetadataWrite
+from cognite.client.exceptions import CogniteAPIError
+
+import cognite.powerops.client._generated.v1.data_classes as v1_data_classes
+from cognite.powerops.resync.v2.data_classes import (
+    FileUploadConfiguration,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def get_type_prefix(domain_model_type: Union[type | str]) -> str:
+    """Get the type prefix for a domain model type.
+
+    Given the domain model type, this function returns the type prefix as a snake case string. Also strips the "Write"
+    suffix from the type name to get the root type name.
+
+    Args:
+        domain_model_type: The domain model type to get the prefix for.
+
+    Returns:
+        The type prefix as a snake case string.
+    """
+
+    if isinstance(domain_model_type, str):
+        type_name = domain_model_type
+    else:
+        type_name = domain_model_type.__name__
+
+    # Convert the type name to snake case
+    external_id_prefix = re.sub(r"(?<!^)(?=[A-Z])", "_", type_name.replace("Write", "")).lower()
```

Comment:
Can we use a separate `to_snake` function instead of using regex directly? Maybe import `to_snake` from `pydantic.alias_generators`?